### PR TITLE
Reformat using clang-format at HEAD.

### PR DESCRIPTION
**Pull request by mprobst**
find modules/angular2 -iname '*.js' | xargs clang-format -i -style="{BasedOnStyle: Google, ColumnLimit: 120}"

NOCOMM: This is not supposed to go in, but serves as a demonstration of clang-format for AtScript.

### CODE REVIEW

**Comment by alexeagle** [comment]
When we are happy with the format, would we want to add it into the angular toolchain and then check in a change to start using it?

**Comment by mprobst** [comment]
I think that should be the plan, once I have the multi-line patch in and the code is ok.

We could integrate it like in Google internally, by requiring all *added* code to be clang-formatted, but leave existing code.

By the way: the build failure is caused by our code missing semicolons, and clang-format incorrectly indenting it. Shouldn't this be caught by some linter? Might actually be useful to have clang-format fail to indent that code just so we notice the missing semicolons.

**Comment by mhevery** [comment]
+1 for failing on semicolon. 
Could you create a PR which corrects the missing `;`?

File: modules/angular2/src/change_detection/abstract_change_detector.js [hunk 1]
```diff
@@ -13,28 +13,20 @@ export class AbstractChangeDetector extends ChangeDetector {
...
-  addChild(cd:ChangeDetector) {
+  addChild(cd: ChangeDetector) {
     ListWrapper.push(this.children, cd);
     cd.parent = this;
   }
 
-  removeChild(cd:ChangeDetector) {
-    ListWrapper.remove(this.children, cd);
-  }
+  removeChild(cd: ChangeDetector) { ListWrapper.remove(this.children, cd); }
```
Comment (mhevery) [inline modules/angular2/src/change_detection/abstract_change_detector.js]: I find it strange that it turned this into a single line.

File: modules/angular2/src/change_detection/coalesce.js [hunk 2]
```diff
@@ -37,52 +37,23 @@ export function coalesce(records:List<ProtoRecord>):List<ProtoRecord> {
...
 
-function _findMatching(r:ProtoRecord, rs:List<ProtoRecord>){
-  return ListWrapper.find(rs, (rr) =>
-    rr.mode === r.mode &&
-    rr.funcOrValue === r.funcOrValue &&
-    rr.contextIndex === r.contextIndex &&
-    ListWrapper.equals(rr.args, r.args)
-  );
+function _findMatching(r: ProtoRecord, rs: List<ProtoRecord>) {
+  return ListWrapper.find(rs, (rr) => rr.mode === r.mode && rr.funcOrValue === r.funcOrValue &&
```
Comment (mhevery) [inline modules/angular2/src/change_detection/coalesce.js]: In this instance I feel like the original was more readable.

File: modules/angular2/src/change_detection/abstract_change_detector.js (hunk 1 above)
Comment (mprobst) [inline modules/angular2/src/change_detection/abstract_change_detector.js]: clang-format in general formats single line functions into single lines. It takes a bit getting used to, but you end up with much more compact code, I personally like it.

File: modules/angular2/src/change_detection/coalesce.js (hunk 2 above)
Comment (mprobst) [inline modules/angular2/src/change_detection/coalesce.js]: Quite possibly. I think the point of an automatic formatter is that if it gets 90% right, it's good enough, and the productivity benefit outweighs any syntax problems. In particular in cases that get formatted less than ideally, if you use clang-format, you'd simply change the code a bit such that the automatic formatting looks good. In this case, e.g. by extracting a local variable or two.

**Comment by mprobst** [comment]
@mhevery: Regarding the semicolons, is there an automatic linter that understands the code base somewhere? The current process of finding the missing semicolons is clang-formatting, then gulp compiling, then inserting the semicolon in the formatted and in the original code, then gulp compiling to get the error from the next file. Turnaround there is about a minute or so, which doesn't seem like a good use of time.

**Comment by mhevery** [comment]
hmm, nothing for AtScript, there are things for JS, Maybe TypeScript will be able to do it.

**Comment by mhevery** [comment]
closing, as I don't think we will get any more value from it here.

**Pull request closed without merging**